### PR TITLE
Everywhere: Some Tuple cleanups and structured bindings for HashMaps

### PR DESCRIPTION
**Pull request by Hendiadyoin1**
### LibTLS: Move Strings when creating RelativeDistinguishedNames

Also cleans up the organizational_unit() helper to use `value_or({})`
instead of doing the same thing manually.

### Kernel/PCI: Prefer structured bindings when iterating over HashMaps

This makes `Access::rescan_hardware` look a bit nicer.

### CODE REVIEW

File: AK/Tuple.h [hunk 1]
```diff
@@ -221,6 +221,26 @@ Tuple(Args... args) -> Tuple<Args...>;
...
 
+namespace std {
+
+template<class T>
+struct tuple_size;
+
+template<size_t index, class T>
+struct tuple_element;
+
+}
```
Comment (ADKaster) [inline AK/Tuple.h:241]: It's not legal to forward declare these...

If the stdlib is libc++, these will actually be declared in `_LIBCPP_NAMESPACE_STD` which defaults to `namespace std { inline namespace __v1 {`, or other names on weird platforms like Android where there's a 'system' libc++ and a 'user' libc++ in different ABI namespaces.

What happens if you compile a source file that includes both `AK/Tuple.h` and `<tuple>` ? with both libc++ and libstdc++. 

Have a look here: https://godbolt.org/z/9c78dE6EY

**Review by ADKaster — CHANGES_REQUESTED** [review]
The two cases where you enabled structured bindings for Tuple should be re-written to use a struct with named items, not a Tuple.

File: AK/Tuple.h [hunk 2]
```diff
@@ -135,6 +129,21 @@ struct Tuple : Detail::Tuple<Ts...> {
...
+    {
+        if (&other == this)
+            return *this;
+        set(other, Indices());
+        return *this;
+    }
+
+    template<typename... Us>
+    Tuple& operator=(Tuple<Us...>&& other)
+    {
```
Comment (ADKaster) [inline AK/Tuple.h:141]: I'd rather not add logic to allow using references in Tuple without a real use case in user/library code that needs it

**Comment by nico** [comment]
The LibPDF commit in here looks good to me.

**Review by ADKaster — APPROVED** [review]

**Pull request merged**